### PR TITLE
repo-sync-2024-07-16T11:00:15+0800

### PR DESCRIPTION
**Pull request by YanZhuangz**
### Added
[Feature] domainRoute now supports setting prefix paths
[Feature] kuscia job now supports multiple parties (1 party , 2 parties , 3 parties)
[Feature] appImage's mountPath now supports relative paths
[Feature] Support for setting DomainRoute reverse tunnel mode through the Kuscia API
[Feature] Support for PKCS 8 format certificates
[Feature] Added DataSource List API to KusciaAPI. KusciaAPI Master side supports Query and List DataSource but does not display sensitive information.

### Changed
[Documentation] Correct description of monitoring metrics in the Kuscia metric documentation
[scripts] During kuscia init, check that the node ID cannot be master
[Feature] Remove the trailing "/" from the RelatedURI field when creating domainData via KusciaAPI
[Feature] Added heartbeat time check to domainRoute, setting it to not ready if it exceeds the threshold
[Feature] Moved error code definitions in the code to Protobuf

### Breaking Changed
[NA]

### Fixed
[Bugfix] Fixed the issue of port conflicts in runp serving multi-replica
[Bugfix] Fixed the issue of state anomalies being caused by multiple executions of kusciaJob stop and restart
[Unit Test] Improved the stability and concurrency of unit tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,58 +37,50 @@ executors:
...
           name: "install go"
           command: |
             ARCH=$(uname -m)
-            case "$ARCH" in
-            x86_64) ARCH="amd64";;
-            aarch64) ARCH="arm64";;
-            *) echo "Unsupported architecture"; exit 1;;
-            esac
 
+            if [ "$ARCH" == "aarch64" ]; then
```
Comment (yushiqie) [inline .circleci/config.yml:49]: 没有这些了吧

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,58 +37,50 @@ executors:
     shell: /bin/bash --login -eo pipefail
 
 commands:
```
Comment (yushiqie) [inline .circleci/config.yml:39]: .circleci 的配置，直接按照main分支copy一份

File: build/dockerfile/kuscia-secretflow.Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-ARG KUSCIA_IMAGE="secretflow/kuscia:0.9.0b0"
+ARG KUSCIA_IMAGE="secretflow/kuscia:latest"
```
Comment (yushiqie) [inline build/dockerfile/kuscia-secretflow.Dockerfile:1]: 这里换成 ‘secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/kuscia:0.10.0b0’

File: build/dockerfile/kuscia-secretflow.Dockerfile [hunk 4]
```diff
@@ -12,11 +12,11 @@ RUN yum install -y protobuf libnl3 libgomp && \
     grep -rl '#!/root/miniconda3/envs/secretflow/bin' /usr/local/bin/ | xargs sed -i -e 's/#!\/root\/miniconda3\/envs\/secretflow/#!\/usr\/local/g' && \
     rm /usr/local/bin/openssl
 
-ARG SF_VERSION="1.7.0b0"
+ARG SF_VERSION="1.6.0b0"
```
Comment (yushiqie) [inline build/dockerfile/kuscia-secretflow.Dockerfile:15]: 这里sf 版本是 1.7.0b0 吧，任务有跑过吗

File: docs/deployment/Docker_deployment_kuscia/deploy_master_lite_cn.md [hunk 5]
```diff
@@ -11,17 +11,21 @@
...
+
+登录到安装 master 的机器上，假设对外 IP 是 1.1.1.1。
+
 指定 Kuscia 版本：
+
 ```bash
-# 使用的 Kuscia 镜像，这里使用 0.9.0b0 版本镜像
-export KUSCIA_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/kuscia:0.9.0b0
+# 使用的 Kuscia 镜像，这里使用 latest 版本镜像
+export KUSCIA_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/kuscia
```
Comment (yushiqie) [inline docs/deployment/Docker_deployment_kuscia/deploy_master_lite_cn.md:21]: 这里使用 0.10.0b0 版本

File: docs/deployment/Docker_deployment_kuscia/deploy_master_lite_cn.md [hunk 6]
```diff
@@ -11,17 +11,21 @@
...
 ```
 
 指定 Secretflow 版本：
+
 ```bash
-# 使用的 Secretflow 镜像，这里使用 1.7.0b0 版本镜像
-export SECRETFLOW_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/secretflow-lite-anolis8:1.7.0b0
+# 使用的 Secretflow 镜像，这里使用 1.6.0b0 版本镜像
+export SECRETFLOW_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/secretflow-lite-anolis8:1.6.0b0
```
Comment (yushiqie) [inline docs/deployment/Docker_deployment_kuscia/deploy_master_lite_cn.md:28]: 这里是 1.7.0b0 版本 sf 吧

File: docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md [hunk 7]
```diff
@@ -16,14 +16,17 @@
 
 登录到安装 alice 的机器上，本文为叙述方便，假定节点 ID 为 alice ，对外可访问的 PORT 是 11080 。
 
-指定 Kuscia 使用的镜像版本，这里使用 0.9.0b0 版本
+指定 Kuscia 使用的镜像版本，这里使用 latest 版本
```
Comment (yushiqie) [inline docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md:19]: 不能使用latest版本，指定具体版本

File: docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md [hunk 8]
```diff
@@ -16,14 +16,17 @@
...
+
 ```bash
-export KUSCIA_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/kuscia:0.9.0b0
+export KUSCIA_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/kuscia
```
Comment (yushiqie) [inline docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md:22]: 这里版本

File: docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md [hunk 9]
```diff
@@ -16,14 +16,17 @@
...
 ```
+
 指定 Secretflow 版本：
+
 ```bash
-# 使用的 Secretflow 镜像，这里使用 1.7.0b0 版本镜像
-export SECRETFLOW_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/secretflow-lite-anolis8:1.7.0b0
+# 使用的 Secretflow 镜像，这里使用 1.6.0b0 版本镜像
+export SECRETFLOW_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/secretflow-lite-anolis8:1.6.0b0
```
Comment (yushiqie) [inline docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md:29]: TODO：这个我们看看抽出两个变量，sf_version 和 kuscia_version. 所有的文档或者脚本中使用到这两个变量的，全局替换。

File: docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md [hunk 10]
```diff
@@ -47,26 +53,28 @@ docker run -it --rm ${KUSCIA_IMAGE} kuscia init --mode autonomy --domain "alice"
 # -m 或者 --memory-limit 参数给节点容器设置适当的内存限制。例如，'-m 4GiB 或 --memory-limit=4GiB' 表示限制最大内存 4GiB，'-m -1 或 --memory-limit=-1'表示没有限制，不设置默认 master 为 2GiB，lite 节点 4GiB，autonomy 节点 6GiB。
 ./kuscia.sh start -c autonomy_alice.yaml -p 11080 -k 11081
 ```
-> 注意事项：<br>
```
Comment (yushiqie) [inline docs/deployment/Docker_deployment_kuscia/deploy_p2p_cn.md:50]: TODO：目前暴露的端口实在是太多了，之后只需要暴露gateway的两个端口，授权端口以及非授权端口，节点应用通过gateway访问

**Review by UniqueMarvin — APPROVED** [review]
LGTM

File: docs/development/register_custom_image.md [hunk 11]
```diff
@@ -9,11 +9,10 @@
 ### 获取工具脚本
 
 ```shell
-# 中心化组网部署模式
-docker cp ${USER}-kuscia-master:/home/kuscia/scripts/tools/register_app_image .
+# 使用的 Kuscia 镜像，这里使用 latest 版本镜像
+export KUSCIA_IMAGE=secretflow-registry.cn-hangzhou.cr.aliyuncs.com/secretflow/kuscia
```
Comment (yushiqie) [inline docs/development/register_custom_image.md:13]: 不要使用任何latest 版本。要使用具体版本

**Review by yushiqie — APPROVED** [review]
LGTM

**Pull request merged**